### PR TITLE
mySolution

### PR DESCRIPTION
**Pull request by AntonFof**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let separatorSign = separator;
+
+    if (separator === undefined) {
+      separatorSign = ',';
+    }
```
Comment (sTorba24) [inline src/arrayMethodJoin.js:13]: you can set it as default value in arguments

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
```
Comment (sTorba24) [inline src/arrayMethodJoin.js:7]:
```suggestion
  [].__proto__.join2 = function(separator = ',') {
```

File: src/arrayMethodJoin.js (hunk 1 above)
Comment (sTorba24) [reply src/arrayMethodJoin.js:13]: remove it

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -5,7 +5,27 @@
...
+      separatorSign = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (item === undefined || item === null) {
+        item = '';
+      }
+      result += item;
```
Comment (sTorba24) [inline src/arrayMethodJoin.js]:
```suggestion
      if (item !== undefined && item !== null) {
         result += this[i];      
       }
```

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -5,7 +5,27 @@
...
-    // write code here
+    let result = '';
+    let separatorSign = separator;
+
+    if (separator === undefined) {
+      separatorSign = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
```
Comment (sTorba24) [inline src/arrayMethodJoin.js]: remove

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Almost good ! lets improve it

**Comment by AntonFof** [comment]
Changes done

**Pull request closed without merging**